### ISSUE TITLE
Module request: `libmultiprocess`

### ISSUE THREAD

**Issue by plebhash** [NixOS/nixpkgs#359902]
## Description

`libmultiprocess` is a C++ library and code generator making it easy to call functions and reference objects in different processes.

Bitcoin Core is starting to use this lib as dependency for IPC.

## Notify maintainers

@prusnak @roconnor

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by plebhash** [comment]
cc @ryanofsky

**Comment by prusnak** [comment]
Can you try to come up with a PR? Should not be that hard.

Tag me in the PR so I can review it.

**Comment by plebhash** [comment]
> Can you try to come up with a PR? Should not be that hard.
> 
> Tag me in the PR so I can review it.

I agree it should be relatively straightforward, but recently I've been a bit busy with Stratum V2 development.

I'll give it a try whenever I get some free time, in case someone else doesn't do it first.

My main motivation is to add some IPC-enabled builds to my [nix-bitcoin-core-archive](https://github.com/plebhash/nix-bitcoin-core-archive)
